cli/command/container/attach.go

```diff
@@ -105,7 +105,12 @@ func RunAttach(ctx context.Context, dockerCLI command.Cli, containerID string, o
 
 	if opts.Proxy && !c.Config.Tty {
 		sigc := notifyAllSignals()
-		go ForwardAllSignals(ctx, apiClient, containerID, sigc)
+		// since the ForwardAllSignals already registers its own signal handler
+		// and should notify the daemon independently of the clients ctx cancellation
+		// we should use a background context to avoid the ForwardAllSignals from
+		// returning before all signals are forwarded to the daemon
+		bgCtx := context.WithoutCancel(ctx)
```

Comment:
nit: since it's not a background context anymore, we should adjust the naming and the comment above. Also, since ForwardAllSignals seems to check for ctx.Done(), maybe we could still use a timeout with this context

Reply:
It's difficult to determine when the `ForceAllSignals` should stop forwarding signals based on a timeout. I think the logic is incorrect to also check `ctx.Done()` inside the `ForceAllSignals` func since it will stop the infinite loop once the signal chan is closed which happens once this function returns which happens on ctx cancellation anyway.

Reply:
just for more clarification, for this feature to work properly in its current implementation the ctx should not be cancelled and instead the signal channel should be closed. 

I'm keeping the ctx.Done() check in the `ForceAllSignals` function since some tests rely on this behavior even though it's not used in its actual implementation.